soem/ethercatcoe.c

```diff
@@ -807,12 +808,12 @@ int ecx_readPDOassignCA(ecx_contextt *context, uint16 Slave, int Thread_n,
  * @param[out] Isize   = Size in bits of input mapping (txPDO) found
  * @return >0 if mapping successful.
  */
-int ecx_readPDOmap(ecx_contextt *context, uint16 Slave, int *Osize, int *Isize)
+int ecx_readPDOmap(ecx_contextt *context, uint16 Slave, uint16 *Osize, uint16 *Isize)
```

Comment:
Isoze and Osize per slave can never be more than uint16, but many slaves together can be much larger. So I propose uint32.

Reply:
Sure, and the call to ecx_readPDOmap(CA) from ecx_map_coe_soe  is on a per slave basis and would fit uint16 here?

Reply:
For now PDOmap can not be larger than 64K but the only reason is that there are no slaves on the market with more than 8KB of mappable RAM. In the future there might be slaves with more memory. The memory mapping in an ESC allows for 64K address space, 4K is reserved. 60KB times 8 bits is 480Kbits. Well over uint16.

These kind of code changes tend to stick for a long time, and in the end you run into unexpected behaviour. Int might not be the best type, uint32 would be better. Performance and storage wise this is neutral anyway.

Reply:
sure, we'll go for that